src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -437,6 +441,15 @@ private static GeneratorState SetGeneratorException(CommonMessageProvider provid
             return new GeneratorState(generatorState.Info, e, diagnostic);
         }
 
+        private static string GetDirectoryForGenerator(Compilation compilation, ISourceGenerator generator)
+        {
+            var outputDirectory = compilation.Options.GeneratedFilesOutputDirectory;
+            var prefix = GetFilePathPrefixForGenerator(generator);
+            return outputDirectory is null
```

Comment:
I'm not sure if this can happen?

Reply:
I agree it shouldn't happen but rather than hedge and silently fail here I'd prefer a `Debug.Assert` verifying it on the method.

Reply:
> I agree it shouldn't happen but rather than hedge and silently fail here I'd prefer a `Debug.Assert` verifying it on the method.

Fixed.